dnf5/commands/repoquery/repoquery.cpp

```diff
@@ -32,12 +32,32 @@ along with libdnf.  If not, see <https://www.gnu.org/licenses/>.
 #include <libdnf5/utils/bgettext/bgettext-mark-domain.h>
 #include <libdnf5/utils/patterns.hpp>
 
-#include <iostream>
-
 namespace dnf5 {
 
 using namespace libdnf5::cli;
 
+template <typename F>
+libdnf5::rpm::PackageQuery get_all_by_function(
+    F & filter_by_candidates, libdnf5::rpm::PackageQuery & candidates, const std::vector<std::string> & arches) {
```

Comment:
Is the `template` necessary here? Why not use just:
```cpp
libdnf5::rpm::PackageQuery get_all_by_function(
    std::function<void(libdnf5::rpm::PackageQuery &, const libdnf5::rpm::PackageQuery &)> filter_by_candidates, 
    libdnf5::rpm::PackageQuery & candidates, const std::vector<std::string> & arches);
```

Reply:
That looks better, thanks. :slightly_smiling_face: